app/src/main/java/io/gnosis/safe/ui/base/BaseActivity.kt

```diff
@@ -12,8 +12,8 @@ import io.gnosis.safe.di.modules.ViewModule
 abstract class BaseActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
         HeimdallApplication[this].inject(this)
+        super.onCreate(savedInstanceState)
```

Comment:
Are you sure about this? :D 
I learned `super.onCreate(...)` should always be done first ... more or less :P

Reply:
> When using activities, inject Dagger in the activity's onCreate() method before calling super.onCreate() to avoid issues with fragment restoration. During the restore phase in super.onCreate(), an activity attaches fragments that might want to access activity bindings.

https://developer.android.com/training/dependency-injection/dagger-android

Reply:
I have no problem reverting that though, if you feel like it could be a source of problems

Reply:
if they recommend it should be ok ... right? ... they are google ... :P

Reply:
I don't trust them fully :) in the case for activity I see no harm. For fragments though, I think you have a point. But maybe it also doesn't matter if the factories get called again.